rimage/image_file.go

```diff
@@ -400,38 +400,65 @@ func IsImageFile(fn string) bool {
 
 // ImageToUInt8Buffer reads an image into a byte slice in the most common sense way.
 // Left to right like a book; R, then G, then B. No funny stuff. Assumes values should be between 0-255.
-func ImageToUInt8Buffer(img image.Image) []byte {
+// if  changeToBGR is true, then R and B get swapped.
+func ImageToUInt8Buffer(img image.Image, changeToBGR bool) []byte {
 	output := make([]byte, img.Bounds().Dx()*img.Bounds().Dy()*3)
 	for y := 0; y < img.Bounds().Dy(); y++ {
 		for x := 0; x < img.Bounds().Dx(); x++ {
 			r, g, b, a := img.At(x, y).RGBA()
 			rr, gg, bb, _ := rgbaTo8Bit(r, g, b, a)
-			output[(y*img.Bounds().Dx()+x)*3+0] = rr
-			output[(y*img.Bounds().Dx()+x)*3+1] = gg
-			output[(y*img.Bounds().Dx()+x)*3+2] = bb
+			if changeToBGR {
+				output[(y*img.Bounds().Dx()+x)*3+0] = bb
+				output[(y*img.Bounds().Dx()+x)*3+1] = gg
+				output[(y*img.Bounds().Dx()+x)*3+2] = rr
+			} else {
+				output[(y*img.Bounds().Dx()+x)*3+0] = rr
+				output[(y*img.Bounds().Dx()+x)*3+1] = gg
+				output[(y*img.Bounds().Dx()+x)*3+2] = bb
+			}
 		}
 	}
 	return output
 }
 
 // ImageToFloatBuffer reads an image into a byte slice (buffer) the most common sense way.
 // Left to right like a book; R, then G, then B. No funny stuff. Assumes values between -1 and 1.
-func ImageToFloatBuffer(img image.Image) []float32 {
+// if  changeToBGR is true, then R and B get swapped.
+// if meanValue and stdDev are not length 0, use those instead to normalize the bytes.
+func ImageToFloatBuffer(img image.Image, changeToBGR bool, meanValue, stdDev []float32) []float32 {
 	output := make([]float32, img.Bounds().Dx()*img.Bounds().Dy()*3)
+	if len(meanValue) == 0 {
+		meanValue = []float32{0.5, 0.5, 0.5}
```

Comment:
Are we thinking people would treat the RGB (or BGR 🙃 ) channels differently? I'm not really seeing it if they're really color images

Reply:
if they don't specify any normalization themselves, I just kept the defaults to be what was already assumed in the function, which is important at least for all the models we currently have running the same as they have